gnomad/resources/grch38/gnomad.py

```diff
@@ -593,46 +518,48 @@ def gnomad_gks_batch(
 
     # Add .vrs and .vrs_json (the JSON string representation of .vrs)
     # Omits .location._id
-    ht_with_gks = add_gks_vrs(ht)
-
-    # If not vrs_only, include the VA freq in various operations below
-    if not vrs_only:
-        # Add .gks_va_freq_dict
-        # Omits .focusAllele
-        ht_with_gks = add_gks_va(
-            ht=ht_with_gks,
-            label_name="gnomAD",
-            label_version=version,
-            coverage_ht=coverage_ht,
-            ancestry_groups=pops_list,
-            ancestry_groups_dict=POP_NAMES,
-            by_sex=by_sex,
-        )
 
-    filtered = hl.filter_intervals(ht_with_gks, [locus_interval])
-    select_cols = {"vrs_json": filtered.vrs_json}
-    if not vrs_only:
-        select_cols["gks_va_freq_json"] = hl.json(filtered.gks_va_freq_dict)
-    annotations = filtered.select(**select_cols).collect()  # might be big
+    # Filter and Select before adding annotations, so that everything can be kept in native python (is that what it's called)
+    filtered = hl.filter_intervals(ht, [locus_interval])
+    # select_cols = {"vrs_json": filtered.vrs_json}
+    
+    struct_list = filtered.collect() # this return a list of structs that can be indexed
+
+    # Add .vrs and .vrs_json (the JSON string representation of .vrs)
+    # Omits .location._id
+    
+    for xi in range(len(struct_list)):
+        struct_list[xi] = add_gks_vrs_py(struct_list[xi])
+        if not vrs_only:
+            struct_list[xi] = add_gks_va_py(
+                            input_dict=struct_list[xi],
+                            label_name="gnomAD",
+                            label_version=version,
+                            coverage_ht=coverage_ht,
```

Comment:
add_gks_va_py is currently dependent on running add_gks_vrs_py because it requires the presence of "original_struct" can we drop the use of original_struct  throughout the functions, since all the information is already in struct_list?